core/system/src/unvic.c

```diff
@@ -91,113 +97,78 @@ static int unvic_acl_check(int irqn)
     /* get vector entry */
     uv = &g_unvic_vector[irqn];
 
-    /* an IRQn slot is considered as registered if the ISR entry in the unprivileged
-     * table is not 0 */
-    is_irqn_registered = uv->hdlr ? 1 : 0;
+    if (uv->id == g_active_box) {
+        return UNVIC_ISR_OWNER_SELF;
+    }
+    if (uv->id == UVISOR_BOX_ID_INVALID) {
+        return UNVIC_ISR_OWNER_NONE;
+    }
+    HALT_ERROR(PERMISSION_DENIED, "Permission denied: IRQ %d is owned by box %d\n\r", irqn, uv->id)
```

Comment:
Now the return values from both `unvic_acl_check` and `unvic_isr_register` do not make sense anymore. I'd rather have `unvic_acl_check` return the ownership status and `unvic_isr_register` be a `void` function that either registers the IRQ or halts.